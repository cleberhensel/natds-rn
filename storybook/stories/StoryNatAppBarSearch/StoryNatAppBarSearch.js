import React from 'react';
import {
  ScrollView,
  SafeAreaView,
  Dimensions,
} from 'react-native';
import {
  NatContainer,
  NatText,
} from '../../../lib';

const styles = {
  defaultScreen: {
    backgroundColor: '#FAFAFA',
    marginTop: 0,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatAppBarSearch = () => (
  <SafeAreaView>
    <NatContainer style={styles.defaultScreen}>
      <ScrollView>
        <NatText value="Developing..." />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>
);

StoryNatAppBarSearch.propTypes = {
};
