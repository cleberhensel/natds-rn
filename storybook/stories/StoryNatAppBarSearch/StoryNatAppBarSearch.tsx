import React, { useState } from 'react';
import {
  ScrollView, SafeAreaView, Dimensions, Alert,
} from 'react-native';
import { NatContainer, NatText, NatAppBarSearch } from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatAppBarSearch: React.SFC<{}> = () => {
  const [search1, setSearch1] = useState('');
  const [search2, setSearch2] = useState('');
  const [search3, setSearch3] = useState('');
  const [search4, setSearch4] = useState('');
  const [search5, setSearch5] = useState('');

  return (
    <SafeAreaView style={styles.defaultScreen}>
      <NatContainer>
        <ScrollView>
          <NatAppBarSearch
            title="Search1"
            value={search1}
            onSubmit={text => setSearch1(text)}
            onPressLeft={() => Alert.alert('press left icon')}
          />
          <NatText value={`Search value: ${search1}`} />
          <NatAppBarSearch
            title="Search2"
            value={search2}
            onSubmit={text => setSearch2(text)}
            onPressLeft={() => Alert.alert('press left icon')}
            onPressRight={() => Alert.alert('press right icon')}
            iconRight={{ svg: { name: 'outlined-action-mic' } }}
          />
          <NatText value={`Search value: ${search2}`} />
          <NatAppBarSearch
            title="Search3"
            value={search3}
            onSubmit={text => setSearch3(text)}
            showIconLeft={false}
            onPressLeft={() => Alert.alert('press left icon')}
            onPressRight={() => Alert.alert('press right icon')}
            iconRight={{ svg: { name: 'outlined-action-mic' } }}
          />
          <NatText value={`Search value: ${search3}`} />
          <NatAppBarSearch
            title="Search4"
            value={search4}
            onSubmit={text => setSearch4(text)}
            showIconLeft={false}
            onPressLeft={() => Alert.alert('press left icon')}
            onPressRight={() => Alert.alert('press right icon')}
          />
          <NatText value={`Search value: ${search4}`} />
          <NatAppBarSearch
            title="Search5 - submit on each tap"
            value={search5}
            onSubmit={text => setSearch5(text)}
            showIconLeft={false}
            onPressLeft={() => Alert.alert('press left icon')}
            onPressRight={() => Alert.alert('press right icon')}
            submitOnChange
          />
          <NatText value={`Search value: ${search5}`} />
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
  );
};
