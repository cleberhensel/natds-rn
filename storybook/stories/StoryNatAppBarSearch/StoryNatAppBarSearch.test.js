import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { StoryNatAppBarSearch } from './StoryNatAppBarSearch';

describe('StoryNatAppBarSearch component', () => {
  it('Should render StoryNatAppBarSearch component ', () => {
    const button = renderer.create(
      <StoryNatAppBarSearch />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
