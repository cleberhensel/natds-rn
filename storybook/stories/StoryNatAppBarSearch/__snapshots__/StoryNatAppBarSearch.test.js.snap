// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatAppBarSearch component Should render StoryNatAppBarSearch component  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#FAFAFA",
          "marginTop": 0,
          "width": 750,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          margin={24}
          style={
            Array [
              Object {
                "marginBottom": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <Text
            color="#333333"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
              ]
            }
            type="body1"
          >
            Developing...
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
